src/nodes.coffee

```diff
@@ -1399,7 +1405,8 @@ exports.Class = class Class extends Base
       method.name = new (if methodName.shouldCache() then Index else Access) methodName
       method.name.updateLocationDataIfMissing methodName.locationData
       method.ctor = (if @parent then 'derived' else 'base') if methodName.value is 'constructor'
-      method.error 'Methods cannot be bound functions' if method.bound
+      method.error 'Cannot define a constructor as a bound function' if method.bound and method.ctor
+      # method.error 'Cannot define a bound method in an anonymous class' if method.bound and not @name
```

Comment:
For illustration, this would be option (1) (disallow bound methods in anonymous classes) of the three options I [outlined](https://github.com/coffeescript6/discuss/issues/84#issuecomment-306605288) for dealing with anonymous classes with bound methods